File: backend/app/schemas/documents.py
"""Pydantic schemas for document APIs."""

from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field, HttpUrl


class DocumentResponse(BaseModel):
    id: str
    title: str
    original_filename: str
    content_type: str | None
    extension: str | None
    file_size_bytes: int
    content_hash: str
    scope: str
    status: str
    version: int
    page_count: int | None
    chunk_count: int
    source_url: str | None
    error_message: str | None
    processed_at: datetime | None
    created_at: datetime
    updated_at: datetime
    meta: dict[str, Any] = Field(default_factory=dict)

    model_config = {"from_attributes": True}


class DocumentChunkResponse(BaseModel):
    id: str
    chunk_index: int
    content: str
    page_number: int | None
    token_count: int | None

    model_config = {"from_attributes": True}


class DocumentListResponse(BaseModel):
    items: list[DocumentResponse]
    total: int
    page: int
    page_size: int


class DocumentContentResponse(BaseModel):
    document_id: str
    text: str
    page_count: int | None
    chunk_count: int


class UrlIngestRequest(BaseModel):
    url: HttpUrl
    title: str | None = Field(default=None, max_length=512)


class MessageResponse(BaseModel):
    message: str

File: frontend/src/types/documents.ts
export type DocumentStatus = "pending" | "processing" | "ready" | "failed" | "deleted";

export type Document = {
  id: string;
  title: string;
  original_filename: string;
  content_type: string | null;
  extension: string | null;
  file_size_bytes: number;
  content_hash: string;
  scope: string;
  status: DocumentStatus;
  version: number;
  page_count: number | null;
  chunk_count: number;
  source_url: string | null;
  error_message: string | null;
  processed_at: string | null;
  created_at: string;
  updated_at: string;
  meta: Record<string, unknown>;
};

export type DocumentListResponse = {
  items: Document[];
  total: number;
  page: number;
  page_size: number;
};

export type DocumentContentResponse = {
  document_id: string;
  text: string;
  page_count: number | null;
  chunk_count: number;
};
